pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -144,10 +148,10 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 
 	var (
 		nonTerminatingNamespace = botanist.SeedNamespaceObject.Status.Phase != corev1.NamespaceTerminating
-		cleanupShootResources   = nonTerminatingNamespace && kubeAPIServerDeploymentFound && infrastructure != nil
+		cleanupShootResources   = nonTerminatingNamespace && kubeAPIServerDeploymentFound && (infrastructure != nil || isWorkerless)
```

Comment:
for a workerless shoot infrastructure will always be nil. So if it is `workerless` then also it will cleanup shoot resources and if it is not workerless then also it will clean up shoot resources. What was intended?

Reply:
we need to cleanup the ManagedResources in the controlplane in case of workerless, so it's required when infrastructure doesn't exist too

Reply:
It is not always true, infra can be nil even if it is shoot with workers.